projects/golang/go/Makefile

```diff
@@ -137,6 +137,42 @@ tar-golang-sources:
 	mkdir -p $(VERSION_DIRECTORY)/archives/$(GOOS)/$(ARCH_LOWER)
 	cd /tmp/go-extracted/usr/lib/golang && rm -rf ./pkg/obj && tar czf $(VERSION_DIRECTORY)/archives/$(GOOS)/$(ARCH_LOWER)/$(GIT_TAG).$(GOOS)-$(ARCH_LOWER).tar.gz .
 
+.PHONY: generate-golang-checksum
+generate-golang-checksum: generate-rpm-checksum generate-archive-checksum
+
+.PHONY: generate-rpm-checksum
+generate-rpm-checksum: generate-rpm-arch-checksum generate-rpm-nonarch-checksum
+
+.PHONY: generate-rpm-arch-checksum
+generate-rpm-arch-checksum:
+	cd $(VERSION_DIRECTORY)/rpmbuild/$(RPM_ARCH)
+	for file in $$(find . -type f -name 'golang-*.rpm'); do \
+    	filePath=$${file%/*} ; \
+		fileBase=$${file##*/} ; \
+		fileName=$${fileBase%.*} ; \
+		sha256sum $${file} > "$${filePath}/$${fileName}.sha256" ; \
+	done
+
+.PHONY: generate-rpm-nonarch-checksum
+generate-rpm-nonarch-checksum:
+	cd $(VERSION_DIRECTORY)/rpmbuild/noarch
+	for file in $$(find . -type f -name 'golang-*.rpm'); do \
+    	filePath=$${file%/*} ; \
+		fileBase=$${file##*/} ; \
+		fileName=$${fileBase%.*} ; \
```

Comment:
one small change and this is good to go -- let's not remove the file extension from the file name prior to appending the `.sha256`, for both the archives and the RPMs

The file name should be:
`golang-1.16.15-3.amzn2.eks.x86_64.rpm.sha256`
rather than
`golang-1.16.15-3.amzn2.eks.x86_64.sha256`

and 
`go1.16.15.linux-amd64.tar.gz.sha256`
rather than
`go1.16.15.linux-amd64.tar.sha256`